### PR TITLE
Added models and routes for project domains and projects

### PR DESCRIPTION
**Pull request by Saif807380**
- Added schemas for domain model and project model
- Added routes for CRUD operations on domain and project models
- Changed middleware name from `event` to `user`

This will help in finishing #51

### CODE REVIEW

File: server/src/models/Domain.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    imageUrl: {
+        type: String,
+        required: true
+    },
+    projects: {
+        type: [{
+          type: mongoose.Schema.Types.ObjectId,
+          ref: 'project'
+        }],
+        required: true
```
Comment (ShubhankarKG) [inline server/src/models/Domain.js:21]: If domains and projects both are required at the same time, then we will not be able to insert data first from the API. I think required can be false here.

File: server/src/controllers/ProjectController.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  async createProject(req,res){
+    try{
+      // assumes that domain of the project already exists
+      let project = await Project.create(req.body);
+      for(const domainId of project.domains){
+        let domain = await Domain.findById(domainId);
+        domain.projects.push(project._id);
+        await domain.save();
+      }
+      res.status(200).json({ id: project._id });
```
Comment (ShubhankarKG) [inline server/src/controllers/ProjectController.js:59]: 201 for new resource created

Comment (ShubhankarKG) [inline server/src/controllers/ProjectController.js:59]: 201 for new resource created

**Review by ShubhankarKG — CHANGES_REQUESTED** [review]
Thanks! Please ensure that we use `const` at places where the variable won't be modified.

File: server/src/models/Domain.js (hunk 1 above)
Comment (Saif807380) [reply server/src/models/Domain.js:21]: I'll set `required` to `false` and `default` to an empty array

File: server/src/controllers/ProjectController.js (hunk 2 above)
Comment (Saif807380) [reply server/src/controllers/ProjectController.js:59]: I'll make the change

**Review by ShubhankarKG — APPROVED** [review]
LGTM, Thanks.

**Pull request merged**